### PR TITLE
Correct error strings and provide source code for exported items.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: hchiramm <hchiramm@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

### CODE REVIEW

**Comment by humblec** [comment]
@travisn ptal :)

File: cmd/rook/ceph/agent.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func startAgent(cmd *cobra.Command, args []string) error {
 	agent := agent.New(context)
 	err = agent.Run()
 	if err != nil {
-		rook.TerminateFatal(fmt.Errorf("failed to run rook ceph agent. %+v\n", err))
+		rook.TerminateFatal(fmt.Errorf("failed to run rook ceph agent. %+v", err))
```
Comment (travisn) [inline cmd/rook/ceph/agent.go]: why remove the newline? just a minor thing, but seems like it made the pod logs look better

File: cmd/rook/ceph/mon.go [hunk 2]
```diff
@@ -115,7 +115,7 @@ func compareMonSecret(secret, configDir string) error {
 		return nil
 	}
 	if cachedKeyring.Value() != secret {
-		return fmt.Errorf("The keyring does not match the existing keyring in %s. You may need to delete the contents of dataDirHostPath on the host from a previous deployment.", cachedKeyringFile)
+		return fmt.Errorf("keyring does not match the existing keyring in %s. You may need to delete the contents of dataDirHostPath on the host from a previous deployment", cachedKeyringFile)
```
Comment (travisn) [inline cmd/rook/ceph/mon.go]: why change the complete sentences?

File: cmd/rook/ceph/osd.go [hunk 3]
```diff
@@ -206,21 +206,21 @@ func writeOSDConfig(cmd *cobra.Command, args []string) error {
 
 	clientset, _, _, err := rook.GetClientset()
 	if err != nil {
-		rook.TerminateFatal(fmt.Errorf("failed to init k8s client. %+v\n", err))
+		rook.TerminateFatal(fmt.Errorf("failed to init k8s client %+v", err))
```
Comment (travisn) [inline cmd/rook/ceph/osd.go]: The pattern rook follows for error messages is that the error is appended as a new sentence so we need the period.

**Comment by humblec** [comment]
> Can you combine the PRs into a single PR? They're all related and this will lower the burden on the CI system. I'd suggest no need to have separate commits for each storage provider if it's all related changes to error messages and comments formatting.
>

Done and its available at https://github.com/rook/rook/pull/2511

File: cmd/rook/ceph/agent.go (hunk 1 above)
Comment (humblec) [reply cmd/rook/ceph/agent.go]: @travisn  go linter give errors if the error strings are ended with newline or punctuation. thats why its removed in this PR.

File: cmd/rook/ceph/mon.go (hunk 2 above)
Comment (humblec) [reply cmd/rook/ceph/mon.go]: @travisn go linter gives errors when error strings are started with capital or end with punctuation.

File: cmd/rook/ceph/osd.go (hunk 3 above)
Comment (humblec) [reply cmd/rook/ceph/osd.go]: In kubernetes we follow a common pattern for error strings like , for example , above will transform to,
"failed to init k8s client, err: %+v", err"

I can change to this format if are in agreement in a follow up PR.

**Comment by humblec** [comment]
@travisn please see my justification and share your thought :)

**Comment by travisn** [comment]
A couple years ago we started to go through the go linter errors. What we found is that many of the suggestions are suboptimal or don't make sense. Without the [ability to suppress errors](https://github.com/golang/lint/issues/263#issuecomment-363652697) in go linter, we realized it was not worth following the guidance if we couldn't get to a clean state.

The current error message patterns work well for rook. I disagree that error messages shouldn't be complete sentences. Some errors are multiple sentences so they need to be formatted the way they are. Since rook is at the application layer instead of a library, it works very nicely to aggregate error messages as sentences. It translates into a sort of stack trace so you can figure out where an error came. 

So unless there is a real benefit to go linter suggestions I'm suggesting let's not fix them.

**Comment by travisn** [comment]
So this PR can be closed since #2511 was opened?

**Comment by humblec** [comment]
Yeah, closing this one.

**Pull request closed without merging**